### ISSUE TITLE
latest updates break installer / require

### ISSUE THREAD

**Issue by AlmirKadric** [componentjs/builder.js#146]
With the latest updates, the builder no longer can require dependencies accordingly. Or rather so requires by component name in component.json file is no longer working.

The installation directories of dependencies in component.json currently correlates to that of the repository (remote) lookup URL. i.e. "component/indexof" will be installed to ./components/component-indexof. However the name of this component is actually "indexof" as per its component.json file. If you were to require('indexof') it would complain saying that the component could not be required.

The fact that the dependency names, component names and component folders do not correlate is the cause of the problem.

Currently this can be fixed via one of the following options:
- update builder to use component name from component.json instead of folder name during registration (this can be done in the builder on line 829).
- fix the "rewrite requires for dependencies" section in the builder (line 839). This seems to be the place this should get done, but seems to work only under certain circumstances (i.e. worked in our loader but not elsewhere). This requires a bit more investigating.
- update the installer to normalize the dependency name for the folder name used in component.json (this would make it the same as node_modules folder) and builder to normalize the dependency name as well when looking for it

Could you please let me know what you would like done here, I will make the contribution if I know what course of action to take.

Best Regards
Almir Kadric

**Comment by AlmirKadric** [comment]
After taking a look at this, it seems you actually rewrite the dependencies for some of these as well to try and get the correct name. Fixing this would also be another possible solution to this problem.

UPDATE: i put this in the above description

**Comment by ericgj** [comment]
I think TJ reverted the recent change - #144 ?

**Comment by jonathanong** [comment]
yeah its reverted. reopen if it's still an issue with latest version

**Comment by AlmirKadric** [comment]
I tried this with latest master and also 0.11.2, still have the same problem

**Comment by AlmirKadric** [comment]
thanks @jonathanong
are you sure the reversion was published? last commit was authored 8 days ago

**Comment by tj** [comment]
it was reverted in component(1), not this repo, we're going to basically rewrite this entire thing but I'll unpublish and revert in this repo tomorrow

**Comment by AlmirKadric** [comment]
seems the issue here is that there is an assumption that the repository name for the component will be the same as the name of the component.

You have this repo "ReopOwner/node-componentName" but the component is called "componentName" inside it's component.json file. When you require it as such builder is expecting "node-componentName".

After tracing this, seems that the "rewrite requires for dependencies" section and "canonicalized" is responsible for fixing this. However it is not using the name of the component from component.json for it's match, causing it to fail.

From what I have seen of the code so far, the intention would be to fix this here.

**Comment by AlmirKadric** [comment]
ah thanks TJ, will take a look at component(1) for now and check to see if it is working. Let me know if there is anything I can do.

Cheers

**Comment by AlmirKadric** [comment]
Where would one find component(1) btw?

**Comment by tj** [comment]
component/component, I just reverted to using an older version of this library
